saleor/graphql/checkout/mutations.py

```diff
@@ -538,13 +599,19 @@ def perform_mutation(cls, _root, info, checkout_id, store_source):
                 )
 
         try:
-            billing_address = order_data["billing_address"]  # type: models.Address
-            shipping_address = order_data["shipping_address"]  # type: models.Address
+            billing_address = order_data["billing_address"]
+            shipping_address = order_data.get("shipping_address", None)
+
+            billing_address = AddressData(**billing_address.as_data())
+
+            if shipping_address is not None:
+                shipping_address = AddressData(**shipping_address.as_data())
+
             txn = gateway_process_payment(
                 payment=payment,
                 payment_token=payment.token,
-                billing_address=AddressData(**billing_address.as_data()),
-                shipping_address=AddressData(**shipping_address.as_data()),
+                billing_address=billing_address,
+                shipping_address=shipping_address,
                 store_source=store_source,
             )
             if txn.is_success and txn.customer_id and user.is_authenticated:
```

Comment:
I know that this is not part of this PR but maybe we could move this outside the try except? Then we don't need to check `txn.is_success`

Reply:
We definitively should almost move everything from it. Will do that 👍